openmc/data/endf.c

```diff
@@ -0,0 +1,39 @@
+#include <stdlib.h>
+
```

Comment:
Given that this code is unusual in that it skips any and all whitespace, and recognizes a `+`/`-` as occurring after an `e` even when the `e` is missing, some documentation up top of the valid inputs may be warranted. Further, some unit tests would be useful to ensure that the code behaves as expected, and to assist any future developers when making changes. As a suggestion, and also taking into account other observations further below:
```c
/* cfloat_endf
 * Parameters:
 *   const char* buffer   - No NULL-check performed
 *   int n                - No bounds checking performed
 * Notes:
 *   This implementation is meant to only be called by a specific Cython function
 *   and as a result deliberately skips certain checks for performance reasons.
 *   This function will insert an 'e' if it is found to be missing. An internal
 *   buffer-overrun is possible with deliberately crafted input, but is not a
 *   concern within the known execution context.
 */
```